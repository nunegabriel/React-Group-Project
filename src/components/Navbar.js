/* eslint-disable react/no-unescaped-entities */
import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../images/planet.png';

const Navbar = () => {
  const activePage = ({ isActive }) => ({
    textDecoration: isActive ? 'underline' : 'none',
  });

  return (
    <header className="navbar">
      <img src={logo} className="logo" alt="logo" />
      <h1 className="logo-title"> Space Travelers' Hub </h1>
      <nav>
        <ul className="NavLinks">
          <li>
            <NavLink to="/" end style={activePage}>
              Rockets
            </NavLink>
          </li>
          <li>
            <NavLink to="/missions/" style={activePage}>
              Missions
            </NavLink>
          </li>
          <li>
            <NavLink to="/my-profile/" style={activePage}>
              My Profile
            </NavLink>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Navbar;
