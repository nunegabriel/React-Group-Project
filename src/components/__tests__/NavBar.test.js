import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Navbar';

test('renders the navbar component', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
