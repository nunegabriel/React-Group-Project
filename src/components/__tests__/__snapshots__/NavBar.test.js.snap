// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the navbar component 1`] = `
<DocumentFragment>
  <header
    class="navbar"
  >
    <img
      alt="logo"
      class="logo"
      src="planet.png"
    />
    <h1
      class="logo-title"
    >
       Space Travelers' Hub 
    </h1>
    <nav>
      <ul
        class="NavLinks"
      >
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            style="text-decoration: underline;"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class=""
            href="/missions/"
            style="text-decoration: none;"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class=""
            href="/my-profile/"
            style="text-decoration: none;"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
